### ISSUE TITLE
Questions and Improvements?

### ISSUE THREAD

**Issue by lthiery** [adafruit/Adafruit_Sensor#2]
Hello!

I love the idea of this type of abstraction library and it looks like it has a lot of great things going for it. I'm a little confused at the moment about certain things.

1) Why the finite list of sensors types?
Can't we just store sensors in an array and provide some compiler definitions to get to friendly named ones?
For example:
# define LIGHT sensor[0]

And to event.LIGHT() would call it instead

The reason is, what if I have sensors that aren't on this list? What if I have multiples of one sensor? Sure I could easily modify this library to accommodate that but shouldn't an abstraction library not need to be updated per application?

I am sure this has a large part to do with the SIs but that seems more like something that needs to be a convention for people writing the drivers. It can't really be enforced by explicitly naming the sensors.

2) Can we have a single location with sensor_ids and what sensor has taken it?
If I am about to create a driver for a new sensor, I want to make sure the sensor_id is not taken! We could just keep track of this in the README perhaps...

Anyway, I'm really glad there's some people thinking about this. Maybe my desires for a library are a little different then what this library intends on having here. 

Another thing I am interested in is decoupling getEvent() into: startEvent(), getEventRaw() and getEvent(). This would be useful so that a microcontroller could launch many sensor conversion in a series without waiting to hear back from them. Data could be transmitted in raw form offloading the parsing to SI to another system. getEvent() could still be there for those who just want the parsed form as soon as it can be available.

**Comment by bitogre** [comment]
I agree that combining multiple sensor data types is clunky and, in my case, reduces its usability.  It is extra data that is being force upon me to move around. Then there is the issue of covering to Floats and converting to undesired units (m/s^2 vs g-force) causes extra computations that have to be undone and use up valuable clock cycles (even more problematic on something like a Arduino Uno or other 8bit micro-controller).

I am not sure if I should do a complete re-write for embedded application with limits on CPU and memory or add additional methods that allow me to get at the intermediate data before it is converted to the unified units.  I am leaning more in the direction of adding additional methods but I am open to suggestions.

**Comment by cdolan92** [comment]
Any plans to support the LIS3DH board in this project? It seems to be pretty popular and low cost. (Here is a link to the Library)[https://github.com/adafruit/Adafruit_LIS3DH]
